pkg/danm/danm.go

```diff
@@ -183,72 +189,98 @@ func extractConnections(args *cniArgs) error {
   return nil
 }
 
-func setupNetworking(args *cniArgs) (*current.Result, error) {
-  syncher := syncher.NewSyncher(len(args.interfaces))
-  for nicId, nicParams := range args.interfaces {
-    nicParams.DefaultIfaceName = "eth" + strconv.Itoa(nicId)
-    go createInterface(syncher, nicParams, args)
+func getAllocatedDevices(args *cniArgs, devicePool string)(map[string][]string, error){
+  checkpoint, err := checkpoint_utils.GetCheckpoint()
+  if err != nil {
+    return nil, errors.New("failed to instantiate checkpoint object due to:" + err.Error())
   }
-  err := syncher.GetAggregatedResult()
-  return syncher.MergeCniResults(), err
+  resourceMap, err := checkpoint.GetComputeDeviceMap(string(args.podUid))
+  if err != nil || len(resourceMap) == 0 {
+    return nil, errors.New("failed to retrieve Pod info from checkpoint object due to:" + err.Error())
+  }
+  allocatedDevices := make(map[string][]string)
+  for resourcename, resources := range resourceMap {
+    if strings.HasPrefix(resourcename, devicePool) {
+      hostDevice, err := sriov_utils.GetPfName(resources.DeviceIDs[0])
+      if err != nil {
+        return nil, errors.New("failed to get the name of the sriov PF for device "+ resources.DeviceIDs[0] +" due to:" + err.Error())
+      }
+      allocatedDevices[hostDevice] = resources.DeviceIDs
```

Comment:
there is a one-to-one mapping between DanmNet, and devicename
once you found the device belonging to the DanmNet, you can break out from the cycle. no need to parse the rest of the checkpoint file

Reply:
Applied